manifests/gateways/istio-ingress/templates/deployment.yaml

```diff
@@ -198,6 +198,10 @@ spec:
 {{- else }}
 {{ toYaml .Values.global.defaultResources | indent 12 }}
 {{- end }}
+          securityContext:
```

Comment:
Somehow this is what is making your tests fail. I think this is because the proxy needs to listen on ports below 1024, so you need to give that capability to the Envoy proxy in this `securityContext`, which wasn't needed when running with UID 0, cf. [what we optionally do for `istio-proxy` in injected containers](https://github.com/istio/istio/blob/master/manifests/istio-control/istio-autoinject/files/injection-template.yaml#L328):
```
      capabilities:
        add:
        - NET_BIND_SERVICE
        drop:
        - ALL
```

This change is only necessary for gateways.
I would do the same for egress gateway.

Reply:
That definitely makes sense. Will do that.
I'm still suspicious of the filesystem permissions, though. I ran `make docker.proxyv2` and inspecting the resulting image with `dive` I see that `/etc/istio` has owner 0, group 0 and permissions `drwxr-xr-x` so it's perhaps not surprising that when running as 1337 it can't write files into `/etc/isto/certs/`. Perhaps a solution would be to mount an emptyDir there?

Reply:
Yes good idea.

Reply:
The volume `istio-certs` is mounted on `/etc/certs`. What would be an appropriate name for a volume to mount at `/etc/istio/certs`?!

Reply:
Or just RUN chmod in the Dockerfile.

Reply:
Realised I'd got the path wrong: `/etc/istio/proxy`, not `/etc/istio/certs`. I'll add a `proxy-certs` volume and see if it works.

Reply:
Decided to leave the gateways alone, as they ended up being more complicated (see [this comment](https://github.com/istio/istio/pull/20311#issuecomment-581204542))